NitroxClient/GameLogic/Building.cs

```diff
@@ -1,213 +1,1326 @@
+        public void Base_ClearGeometry_Pre(Base @base)
+        {
+            // Base_ClearGeometry_Pre is called twice. One time for the ghost layout and one time for the viewable layout. We are only interested in the viewable layout. 
+            {
+                idCacheForGeometryUpdate.Clear();
+
+                Transform[] cellObjects = (Transform[])@base.ReflectionGet("cellObjects");
+
+                if (cellObjects == null)
+                {
+                    return;
+                }
+
+                foreach (Transform cellObject in cellObjects)
+                {
+                    if (cellObject != null)
+                    {
+                        for (int i = 0; i < cellObject.childCount; i++)
+                        {
+                            Transform surface = cellObject.GetChild(i);
+
+                            if (surface != null && surface.gameObject != null)
+                            {
+
+#if TRACE && BUILDING
+                                string info = generateGameObjectTransformInfo(surface);
+                                NitroxModel.Logger.Log.Debug("Base_ClearGeometry_Pre - current clear for : " + info);
+#endif
+                                if (!surface.name.Contains("CorridorConnector"))
+                                {
+
+                                    NitroxId id = NitroxEntity.GetIdNullable(surface.gameObject);
+
+                                    // Have to ignore BaseRoomCorridorConnector, because they are automatically spawned in case of adding a Corridor or a Hatch to a Room
+                                    if (id != null)
+                                    {
+
+#if TRACE && BUILDING
+                                        NitroxModel.Logger.Log.Debug("Base_ClearGeometry_Pre - saving id for : " + info + " id: " + id);
+#endif
+
+                                        string parentCellContent = string.Empty;
+                                        if (surface.GetSiblingIndex() == 0)
+                                        {
+                                            for (int j = 0; j < surface.parent.childCount; j++)
+                                            {
+                                                if (surface.gameObject != null)
+                                                {
+                                                    parentCellContent = parentCellContent + surface.parent.GetChild(j).name;
+                                                }
+                                            }
+                                        }
+
+                                        GeometryInfo geometryInfo = new GeometryInfo() { Name = surface.name, Id = id, CellPosition = surface.parent.position, CellIndex = surface.GetSiblingIndex(), CellContent = parentCellContent };
+                                        idCacheForGeometryUpdate.Add(geometryInfo);
+                                        NitroxEntity.RemoveId(surface.gameObject);
```

Comment:
I have noticed that Objects are not properly automaticaly destroyed by the game as long as there is a NitroxEntity applied to it. Destroying the object on our own also caused some wired effects. So i thought the best way is, to let the gameengine or unityengine do what it wants to do with the object and just revert the changes that nitrox did to the gameobject (adding a NitroxEntity) and so everything works fine. 
You can see this also if you apply a NitroxId to a ghost object. These objects are also not destroyed because the engine seems to perform a check if any component is still attached and so the NitroxEntity blocks this. I don't know if this is the same for all other entitys but for buildings i noticed this behaviour.